packages/client/src/components/UnprotectedRoute.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Spinner } from '@opencrvs/components'
+import { useDispatch, useSelector } from 'react-redux'
+import { configAnonymousUserLoaded } from '@client/offline/actions'
+import { referenceApi } from '@client/utils/referenceApi'
+import { IStoreState } from '@client/store'
+import { Route, RouteProps } from 'react-router'
+
+const StyledSpinner = styled(Spinner)`
+  position: absolute;
+  margin-left: -24px;
+  margin-top: -24px;
+  top: calc(50% - 20px);
+  left: 50%;
+  width: 40px;
+  height: 40px;
+`
+
+const UnprotectedRouteWrapper = (props: RouteProps) => {
+  const dispatch = useDispatch()
+  const offlineData = useSelector((store: IStoreState) => {
+    return store?.offline?.offlineData.anonymousConfig
+  })
+
+  React.useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { config } = await referenceApi.loadConfigAnonymousUser()
+        if (config) {
+          document.title = config?.APPLICATION_NAME as string
+          dispatch(configAnonymousUserLoaded({ anonymousConfig: config }))
+        }
+      } catch (e) {}
```

Comment:
why don't we just throw this if an error occurs?

Reply:
we tried this from the Page component. But there are additional layers of security and each error already has a definition of redirecting to the login page. we thought to make it simpler.

Unless I didn't understand. can you give me more explanation?

Reply:
I think what Pyry is referring to is just the `try { ... } catch(e) {}` structure without error handling. As it is now, the (potential) error would be hidden completely. So it would probably be better for us developers and also the user if the exception would be thrown and they would see the default error dialog